pubsub/src/types.rs

```diff
@@ -35,47 +36,68 @@ impl<T: PubSubMetadata> PubSubMetadata for Option<T> {
 }
 
 /// Unique subscription id.
+///
 /// NOTE Assigning same id to different requests will cause the previous request to be unsubscribed.
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub enum SubscriptionId {
-	/// U64 number
-	Number(u64),
+pub enum SubscriptionId<N> {
```

Comment:
Ah, I see what you meant with `Into<u64>` in the earlier comment. Why not require `Into<SubscriptionId>` instead? This together with some extra `From` implementations that we would add here (for numeric types and `String` as you did now) would allow us to keep that configurable without maiking `SubscriptionId` generic.